Shreevaths taking over for Manvitha phase4 project manager dashboard

myeeli opened the pull request:
# Description
<img width="544" height="323" alt="image" src="https://github.com/user-attachments/assets/c2dd23cf-1f69-4ac5-a6f7-f5f5086f1fa1" />

<img width="527" height="115" alt="image" src="https://github.com/user-attachments/assets/10316571-d7bd-4270-b431-b8e3d25a74c4" />


## Related PRS (if any):
This frontend PR is related to the [#1886](https://github.com/OneCommunityGlobal/HGNRest/pull/1886) backend PR.
…

## Main changes explained:


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to localhost:5173/educationportal/pm/dashboard

## Screenshots or videos of changes:
[

https://github.com/user-attachments/assets/974f655c-f02b-4cae-8027-f552f94f1b82

](url)

## Note:
Include the information the reviewers need to know.

myeeli committed:
Impelemented PM educator view

myeeli committed:
Implementing PMNotification Component

myeeli committed:
Enhanced filter and search

myeeli committed:
Integrated API's

myeeli committed:
Made css changes and implemented darkmode

myeeli committed:
Fixed 401 issue

myeeli committed:
Fixed Sned Notification issues

myeeli committed:
Resolved merge conflicts

myeeli committed:
Resolved Merge Conflicts

myeeli committed:
Resolved SonarQubesecurity issues

Anusha-Gali reviewed (changes requested):
Hi Manvitha,

I have reviewed your PR locally and have found the below issue. Please have a look into the steps i follow in the video to be able to recreate the bug.

https://github.com/user-attachments/assets/cf04b050-a053-4564-a6be-b7d8e5503764

VijayAnirudh reviewed (changes requested):
Hello!
1) The project dashboard card is fine.
2) The search box works.
3) The view option works on the educator's card.
4) When I click expand button, even though there are n students under an educator, it displays no students found! 
<img width="1595" height="514" alt="image" src="https://github.com/user-attachments/assets/2ab735dd-c891-4d59-b29f-c30d4c941cb4" />
<img width="1372" height="470" alt="image" src="https://github.com/user-attachments/assets/2e27460f-3485-421b-8e46-4df236a8d117" />
<img width="1578" height="844" alt="image" src="https://github.com/user-attachments/assets/8a09f490-5e0a-4076-b059-bafc275485fd" />

myeeli committed:
fixed review comments

myeeli committed:
fixed review comments

myeeli committed:
fixed review comments

myeeli committed:
fixed lint issues

myeeli committed:
fixed lint issues

myeeli committed:
fixed lint issues

SharadhaKasiviswanathan committed:
Resolve merge conflicts with development branch

SharadhaKasiviswanathan committed:
Fix lint errors, unit tests, and KitchenandInventoryLogin null-check bug

- Update jwt-decode imports to v4 named exports across 6 source files
- Update jwt-decode test mocks to use named export in 7 test files
- Fix null-check bug in KitchenandInventoryLogin.jsx
- Fix authInit.test.js mock to provide named jwtDecode export
- 0 lint errors, 313/313 test files passing, build verified

SharadhaKasiviswanathan committed:
fix(pm-dashboard): resolve merge conflicts and fix SonarCloud issues

SharadhaKasiviswanathan committed:
fix(pm): add type=button to all button elements

kzou55 reviewed (changes requested):
Hi Sharadha, 

I ran the PR locally alongside it's associated backend. 

I can't seem to login no matter what account I try to use. I believe the issue is on the frontend side as I can login using the associated backend and running the development frontend but not vice versa. 

<img width="1914" height="1041" alt="Image" src="https://github.com/user-attachments/assets/29833e9b-bdf6-41f9-9a4c-36efb66a829c" />

kzou55 reviewed (changes requested):
Hi Sharadha, 

I ran the PR and its associated frontend. 

I couldn't verify the implementation as the frontend [pr](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/4340#pullrequestreview-4726282248) is giving me issues logging in.

HemanthNidamanuru reviewed (changes requested):
Hi,

I attempted to test this PR locally by checking out the current branch, installing the dependencies, clearing the site data and cache, and navigating to the Education Portal PM Dashboard.

However, I was unable to log in as an admin user, so I could not verify the changes in this PR.


<img width="1918" height="1085" alt="Image" src="https://github.com/user-attachments/assets/6983ec6f-0545-4d46-a9a4-a32ea61b97b3" />

shreevathsrao committed:
Fix the regressed login Issue

shreevathsrao committed:
Fix failing test cases

shreevathsrao committed:
Revert unwanted changes

shreevathsrao committed:
revert kitchen file

shreevathsrao committed:
remove kitchen file from PR

shreevathsrao committed:
revert unwanted test files

akv-iu reviewed (changes requested):
Tested this locally  a few things work well, one issue found.

## Verified

- The project dashboard card renders correctly
- The search box works
- The "View" option on the educator's card works

## Issue:

- Clicking the expand button shows "No students found," even when the educator has students under them.

<img width="2774" height="1478" alt="Image" src="https://github.com/user-attachments/assets/b5d2ae69-5c50-4374-a9df-fd264f718021" />

<img width="2814" height="1390" alt="Image" src="https://github.com/user-attachments/assets/4a43b33d-cb30-437b-982a-60236d921c69" />

shreevathsrao committed:
Fix merge conflict

shreevathsrao committed:
remove unwanted file

iAbhi001 reviewed (changes requested):
I tested this locally by checking out both backend (#1886) and frontend (#4340) PR branches. 

**Verified:**
- [x] The Project Dashboard card renders correctly with proper layout/styling.
- [x] The search box works as expected to filter educators.
- [x] The "View" option on the educator's card functions properly.

**Issue Found:**
- [ ] **Expand Student List Bug:** Clicking the **Expand** button on an educator card still displays *"No students found"*, even when the educator has students assigned to them under their profile/account. 

Please take a look at the data mapping or API response handling for the expanded student list component. Requesting changes until this is resolved.

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/6c9665d6-c358-4f0a-a935-f44829687e80" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/0d750590-fecd-4beb-9b2a-1907d47a9270" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/86d7fbb5-e04e-4462-a986-d8fcdfbf1232" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/e05bf155-fc0c-4c49-b507-e6505de44d21" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/502ae91e-f9ef-495d-afdf-63493da77e4e" />

shreevathsrao committed:
Merge branch 'development' of into Manvitha-Phase4-ProjectManager-Dashboard

kzou55 reviewed (changes requested):
Hi Shreevaths, 

I ran and tested the PR locally alongside its associated backend. 

## Verified
- Search educators work
- Search students work
- Filter by class dropdown works
- "Expand all" and "Collapse all" button works
- "View" and "Hide" button works for classes
- New Announcement button creates a pop up form and user can select all or individual teachers and can hit cancel or send announcement. 
- Works in Dark Mode

## Testing
- Light Mode
  
  https://github.com/user-attachments/assets/7456860e-d3f1-49c3-8181-a5b7100d7d61
- Dark Mode 

  https://github.com/user-attachments/assets/f0507df8-648e-4d4b-96cf-e743b137216b


## Issues
- New Announcement form issues:
  - Could you clarify if this is what the New Announcment form should look like? Compared it to the initial PR and the one I see is different 
    <img width="1748" height="797" alt="image" src="https://github.com/user-attachments/assets/b450d050-fbe6-4549-8ba6-8a437de016a6" />

  - When page is scrolled up, the nav bar covers the New Announcement form

    <img width="1898" height="1030" alt="Image" src="https://github.com/user-attachments/assets/48d06629-80bb-4085-9eb5-f65dbfe579ea" />

    <img width="1906" height="974" alt="Image" src="https://github.com/user-attachments/assets/1ccb9640-86a1-463a-80f2-dcbeca752333" />

    <img width="717" height="970" alt="Image" src="https://github.com/user-attachments/assets/517c415f-f3b4-4f0b-81de-f461c7a83aaa" />

shreevathsrao committed:
fix: announcement modal

kzou55 reviewed (approved):
Hi Shreevaths, 

I ran and tested the PR locally again alongside its associated backend. 

## Verified 
- Confirmed the changes were made to the New Accountment form

    <img width="1911" height="1034" alt="Image" src="https://github.com/user-attachments/assets/c12d1872-cf92-4db5-b94e-edaee475c606" />

    <img width="907" height="722" alt="Image" src="https://github.com/user-attachments/assets/3a76d059-5eaa-4c6f-939d-46abca69141d" />

    <img width="1262" height="335" alt="Image" src="https://github.com/user-attachments/assets/5fe4d03f-a787-42be-821e-c143e4826e44" />

  - Looks good in Dark Mode
    <img width="914" height="730" alt="Image" src="https://github.com/user-attachments/assets/e746b009-9adf-4a12-947c-ff44f2fb62b6" />

    <img width="1234" height="299" alt="Image" src="https://github.com/user-attachments/assets/5190422b-3c4c-426a-ba7b-d97d3c4bdc2d" />
  - Looks good in smaller screen sizes/mobile view

    <img width="494" height="756" alt="Image" src="https://github.com/user-attachments/assets/f16a9ea0-b22f-4e94-b542-fc56ce842ff9" />
 
    <img width="624" height="947" alt="Image" src="https://github.com/user-attachments/assets/5e0a446c-775f-44a5-bbb1-af84ff7cd4d7" />

- All the other functilniaty works as before
